app.py

```diff
@@ -31,46 +30,176 @@ def init_app(self):
         }
         
         self.engine.register_command("show_in_filesystem", self.show_in_filesystem, p)
-            
+
     def launch(self, path):
-        self.log_debug("Launching file system viewer for folder %s" % path)        
-        
-        # get the setting        
+        """
+        Given a path this method will choose the appropriate way to open that as a folder in the OS's default
+        file manager. Folders will be opened as is
+        :param path:
```

Comment:
missing description

Reply:
we should add a description saying that nothing will happen if the path is invalid.